segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/ZKSegmentContainerMonitor.java

```diff
@@ -175,9 +180,14 @@ private void checkAssignment() {
                 Collection<Integer> containersToBeStopped = CollectionHelpers.filterOut(runningContainers, desiredList);
                 containersToBeStopped = CollectionHelpers.filterOut(containersToBeStopped, containersPendingTasks);
 
-                log.info("Container Changes: Desired = {}, Current = {}, PendingTasks = {}, ToStart = {}, ToStop = {}.",
-                        desiredList, runningContainers, containersPendingTasks, containersToBeStarted,
-                        containersToBeStopped);
+                // Only report if we have any changes or if sufficient time has elapsed since the last report.
+                boolean logReport = !(containersPendingTasks.isEmpty() || containersToBeStarted.isEmpty() || containersToBeStopped.isEmpty());
```

Comment:
As far as I understand, we want to log if there is "any" change in the container assignment or state, right? In this case, we don't want to log if "all" of the collections containing the changes in the containers are empty. Thus, shouldn't the condition be `!(containersPendingTasks.isEmpty() && containersToBeStarted.isEmpty() && containersToBeStopped.isEmpty())`? If not, as long as there is a single empty collection, the log wouldn't be reported (despite there could be changes stored in the rest of collections).

Reply:
Yes, you are right. I need some coffee...